data_override/include/get_grid_version.inc

```diff
@@ -45,8 +44,16 @@ subroutine GET_GRID_VERSION_1_(grid_file, mod_name, domain, isc, iec, jsc, jec,
      call mpp_error(FATAL, 'data_override_mod(get_grid_version_1): Error in opening file '//trim(grid_file))
   endif
 
-  call mpp_get_data_domain(domain, isd, ied, jsd, jed)
   call mpp_get_global_domain(domain, isg, ieg, jsg, jeg)
+  call mpp_get_compute_domain(domain, isc, iec, jsc, jec)
+
+  if (lbound(lon,1).ne.isc .or. lbound(lon,2).ne.jsc) then
+    call mpp_error(FATAL, "get_grid_version_1: Incorrect lower-bound dimensions of lon(isc:,jsc:)")
+  endif
+
+  if (lbound(lat,1).ne.isc .or. lbound(lat,2).ne.jsc) then
+    call mpp_error(FATAL, "get_grid_version_1: Incorrect lower-bound dimensions of lat(isc:,jsc:)")
+  endif
```

Comment:
@J-Lentz I noticed this was failing the GNU CI and looked into it a bit since it was coming from a test I wrote in the coupler subdirectory.

I was confused at first why this check kept failing since the passed in array had the right bounds, but after a little digging I realized that when passed in the array bounds get reset from whatever it may be to `1:<size of the array>`.

I thought it was a bug but apparently that's how fortans supposed to work(?). I think we can either leave the lower bounds in the argument declaration so it can be checked, or we could just check the size here instead.